### DISCUSSION TITLE
Error handle

### DISCUSSION THREAD

**Discussion by ilikosov** [rematch/rematch#845] [Q&A] [answered]
what is the best way to catch error in async effector, like server error on http request?

**Comment by semoal** [comment, marked as answer]
Hello! Thanks for using Rematch =)

My personal recommendation is just to use try catch inside await method, and if catchs enters log something descriptive to a model called "errors", with that method of working you could get "realtime" notifications on your view of any error in your model.

```js

export const someModel = {
    state: {},
    effects: dispatch => ({
        async getSomeInfo(payload, rootState) {
            try {
              await someAsyncReq()
            } catch(err) {
              dispatch.globalErrors.add(err)
            }
        },
    }),
}

```
